pkg/azureutils/azure_disk_utils.go

```diff
@@ -172,6 +229,112 @@ func NormalizeNetworkAccessPolicy(networkAccessPolicy string) (compute.NetworkAc
 	return "", fmt.Errorf("azureDisk - %s is not supported NetworkAccessPolicy. Supported values are %s", networkAccessPolicy, compute.PossibleNetworkAccessPolicyValues())
 }
 
+func ParseDiskParameters(parameters map[string]string) (ManagedDiskParameters, error) {
+	var err error
+	if parameters == nil {
+		parameters = make(map[string]string)
+	}
+
+	diskParams := ManagedDiskParameters{
+		DeviceSettings: make(map[string]string),
+		Incremental:    true, //true by default
+		Tags:           make(map[string]string),
+		VolumeContext:  parameters,
+	}
+	for k, v := range parameters {
+		switch strings.ToLower(k) {
+		case consts.SkuNameField:
+			diskParams.AccountType = v
+		case consts.LocationField:
+			diskParams.Location = v
+		case consts.StorageAccountTypeField:
+			diskParams.AccountType = v
+		case consts.CachingModeField:
+			diskParams.CachingMode = v1.AzureDataDiskCachingMode(v)
+		case consts.ResourceGroupField:
+			diskParams.ResourceGroup = v
+		case consts.DiskIOPSReadWriteField:
+			diskParams.DiskIOPSReadWrite = v
+		case consts.DiskMBPSReadWriteField:
+			diskParams.DiskMBPSReadWrite = v
+		case consts.LogicalSectorSizeField:
+			diskParams.LogicalSectorSize, err = strconv.Atoi(v)
+			if err != nil {
+				return diskParams, fmt.Errorf("parse %s failed with error: %v", v, err)
+			}
+		case consts.DiskNameField:
+			diskParams.DiskName = v
+		case consts.DesIDField:
+			diskParams.DiskEncryptionSetID = v
+		case consts.TagsField:
+			customTagsMap, err := util.ConvertTagsToMap(v)
+			if err != nil {
+				return diskParams, err
+			}
+			for k, v := range customTagsMap {
+				diskParams.Tags[k] = v
+			}
+		case azure.WriteAcceleratorEnabled:
+			diskParams.WriteAcceleratorEnabled = v
+		case consts.MaxSharesField:
+			diskParams.MaxShares, err = strconv.Atoi(v)
```

Comment:
Consider refactoring the code in the case into a single function, e.g. `ParseMaxShares`, that is called both here and by `GetMaxShares`.